astroid/inference.py

```diff
@@ -714,16 +717,17 @@ def _infer_binary_operation(left, right, binary_opnode, context, flow_factory):
 def _infer_binop(self, context):
     """Binary operation inference logic."""
     if context is None:
-        context = contextmod.InferenceContext()
```

Comment:
We still need to build a context here including one for lhs and rhs, so not sure if this change actually helps.